### DISCUSSION TITLE
[Profiler] Roofline Model via Static Analysis

### DISCUSSION THREAD

**Discussion by fpham0701** [cornell-zhang/allo#200] [Projects]
Research Task FA '24. Discussion will list updates

**Comment by chhzh123** [comment]
You can check out more information on the implementation on this webpage:
https://www.cs.cornell.edu/courses/cs6120/2022sp/blog/hcl-mlir/

**Comment by fpham0701** [comment]
Have had a busy week with exams/projects
Overviewed background on webpage and synthesized knowledge of the Roofline Model
Started writing Python code but blocked by no access to servers -- waiting for servers to utilize Allo

**Comment by fpham0701** [comment]
Forked repository and added code
Started implementation of "count" method to return total number of operations of type "op_type"
Blueprinted necessary functions to get operational intensity
Looking at customize.py and other code base to map how to get total number

**Comment by fpham0701** [comment]
Continued working on "count" method to get number of operations. Have been able to get arithmetic operations but not other types.
Ran into some problems but continuing to debug

**Comment by fpham0701** [comment]
Implemented a workable scheduling method to count types of operations
Implemented a wrapper method to calculate OI
Worked on presentation
